server/manager/src/main/java/org/apache/accumulo/master/FateServiceHandler.java

```diff
@@ -779,65 +781,37 @@ private String validateNewNamespaceArgument(ByteBuffer namespaceArg, TableOperat
   /**
    * Create a file on the file system to hold the splits to be created at table creation.
    */
-  private String writeSplitsToFile(final long opid, final List<ByteBuffer> arguments,
+  private void writeSplitsToFile(Path splitsPath, final List<ByteBuffer> arguments,
       final int splitCount, final int splitOffset) throws IOException {
-    String opidStr = String.format("%016x", opid);
-    String splitsPath = getSplitPath("/tmp/splits-" + opidStr);
-    removeAndCreateTempFile(splitsPath);
-    try (FSDataOutputStream stream = master.getOutputStream(splitsPath)) {
-      writeSplitsToFileSystem(stream, arguments, splitCount, splitOffset);
+    FileSystem fs = splitsPath.getFileSystem(master.getContext().getHadoopConf());
+    try (FSDataOutputStream stream = fs.create(splitsPath)) {
+      // base64 encode because splits can contain binary
+      for (int i = splitOffset; i < splitCount + splitOffset; i++) {
+        byte[] splitBytes = ByteBufferUtil.toBytes(arguments.get(i));
+        String encodedSplit = Base64.getEncoder().encodeToString(splitBytes);
+        stream.writeBytes(encodedSplit + '\n');
+      }
     } catch (IOException e) {
-      log.error("Error in FateServiceHandler while writing splits for opid: " + opidStr + ": "
-          + e.getMessage());
+      log.error("Error in FateServiceHandler while writing splits to {}: {}", splitsPath,
+          e.getMessage());
       throw e;
     }
-    return splitsPath;
-  }
-
-  /**
-   * Always check for and delete the splits file if it exists to prevent issues in case of server
-   * failure and/or FateServiceHandler retries.
-   */
-  private void removeAndCreateTempFile(String path) throws IOException {
```

Comment:
Any idea why this temp file was created? Sounds like it was created to help with failover somewhere.